sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/models/RecognizeEntitiesResult.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.textanalytics.models;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.util.IterableStream;
+
+import java.util.ArrayList;
+
+/**
+ * The {@link RecognizeEntitiesResult} model.
+ */
+@Immutable
+public final class RecognizeEntitiesResult extends TextAnalyticsResult {
+    private final CategorizedEntityCollection entities;
+
+    /**
+     * Creates a {@link RecognizeEntitiesResult} model that describes recognized entities result.
+     *
+     * @param id Unique, non-empty document identifier.
+     * @param textDocumentStatistics The text document statistics.
+     * @param error The document error.
+     * @param entities An {@link IterableStream} of {@link CategorizedEntity}.
+     * @param warnings A {@link IterableStream} of {@link TextAnalyticsWarning}.
```

Comment:
It will be good to connect this type with `CategorizedEntityCollection` which also includes warnings in there.
Any reason why this is different than .net here?

Reply:
good catch. I will change to use categorized entity collection instead of two inputs.